Adding Quickfix slash command

bassamsdata opened the pull request:
## Description

This PR adds a `/quickfix` slash command, allowing users to send quickfix list entries to the chat. The command handles different quickfix types with the following logic:


Details about the Mechanism:

- **Files only**: If the quickfix entry is a file (e.g., from a search pattern using grep or file list), the whole file content is sent.
- **Diagnostics (e.g., LSP errors/warnings)**: 
  - For small files (<100 lines), the entire file is sent with a diagnostic summary.
  - For large files, only relevant code context is sent, grouped by symbol (function/method/class) or by proximity (10 lines) if no symbols are found.

If a simpler mechanism is more preferable (e.g., always sending all files regardless of quickfix list type), or if you’d rather not include this in the core plugin, I’m happy to adapt this as an extension instead.

Are you okay with this approach, or would you like any changes?

## Notes about the code:

Moved the `symbols` function that extracts symbols with ranges into the helpers file, so it can be reused in both the quickfix and symbols slash commands.

Thank you


## Related Issue(s)

Related to discussion #843


## Screenshots
https://github.com/user-attachments/assets/f97ddafd-10c5-4f7e-bf59-3cd5d4c912db


## Checklist



- [x] I've read the [contributing](https://github.com/olimorris/codecompanion.nvim/blob/main/CONTRIBUTING.md) guidelines and have adhered to them in this PR
- [ ] I've updated `CodeCompanion.has` in the [init.lua](https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/init.lua#L239) file for my new feature
- [x] I've added [test](https://github.com/olimorris/codecompanion.nvim/blob/main/CONTRIBUTING.md#testing) coverage for this fix/feature
- [x] I've updated the README and/or relevant docs pages
- [x] I've run `make all` to ensure docs are generated, tests pass and my formatting is applied

olimorris commented:
Okay so this is really, really cool. Just tried it out now (after figuring out how to add a "send to quickfix" action to Snacks). It works perfectly. Very, very excited about this.

I've recently made some changes in #1633 regarding how we share files and buffers with an LLM:

````md
<attachment filepath="lua/codecompanion/strategies/chat/agents/tools/helpers/patch.lua">
Here is the content from the file:
```lua
local M = {}

M.FORMAT_PROMPT = "some code"
```
</attachment>
````

This is to bring some uniformity across buffer and file shares. An LLM shouldn't really care and the `<attachment>` tags make it easier to group instructions around.

So long winded way of saying, "can you update the PR for this?"

bassamsdata commented on lua/codecompanion/strategies/chat/helpers.lua:
```diff
@@ -84,4 +86,103 @@ function M.add_image(Chat, image, opts)
+  }
+end
+
+---Extract symbols from a file using TreeSitter
```
Done

bassamsdata commented:
I’ve updated the PR to address all the review comments, and I hope everything is correct.  
I also added tests for the symbols Tree-sitter integration. To be honest, I was avoiding this because when I tried it in my plugin before, Tree-sitter gave me quite a headache, but now it’s done!

Apologies for the force-push, I updated the PR branch with the upstream main and performed a rebase, which caused it. Sorry about that!

olimorris commented on tests/strategies/chat/test_helpers.lua:
```diff
@@ -0,0 +1,181 @@
+        -- Helper function to create test files
+        function _G.create_test_files()
+          -- Lua file with functions
+          vim.fn.writefile({
```
Can we create any test stubs in a temp directory using `vim.fn.tempname()`? My concern is if mini.test errors, we'll be left with lot's of files in the repo.

I've done this just now in #1613.

bassamsdata replied:
Thank you for this and the example, indeed, it’s much better!

olimorris commented:
This is amazing! Great job @bassamsdata.
